include/seqan3/argument_parser/validators.hpp

```diff
@@ -297,68 +285,350 @@ class file_ext_validator
+     */
+    bool filename_is_writable(std::filesystem::path const & path) const
+    {
+        std::ofstream file{path};
+        bool is_open = file.is_open();
+        bool is_good = file.good();
+        file.close();
+
+        // Try to remove file.
+        std::filesystem::remove(path);
+
+        return is_open && is_good;
+    }
+
+    //!\brief Stores the extensions.
+    std::vector<std::string> extensions{};
+};
+
+/*!\brief A validator that checks if a given path is a valid input file.
+ * \ingroup argument_parser
+ * \implements seqan3::validator_concept
+ *
+ * \details
+ *
+ * On construction, the validator can receive a list (std::vector over std::string) of valid file extensions.
+ * The struct acts as a functor that throws a seqan3::parser_invalid_argument exception whenever a given filename's
+ * extension (sts::filesystem::path) is not in the given list of valid file extensions, if the file does not exist, or
+ * if the file does not have the proper read permissions.
+ *
+ * \snippet test/snippet/argument_parser/validators_input_file.cpp usage
+ *
+ * \note The validator works on every type that can be implicitly converted to std::filesystem::path.
+ */
+class input_file_validator : public file_validator_base
+{
+public:
+    // Import from base class.
+    using file_validator_base::value_type;
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    input_file_validator() = default;                                         //!< Defaulted.
+    input_file_validator(input_file_validator const &) = default;             //!< Defaulted.
+    input_file_validator(input_file_validator &&) = default;                  //!< Defaulted.
+    input_file_validator & operator=(input_file_validator const &) = default; //!< Defaulted.
+    input_file_validator & operator=(input_file_validator &&) = default;      //!< Defaulted.
+    ~input_file_validator() = default;                                        //!< Defaulted.
+
+    // Import base class constructor.
+    using file_validator_base::file_validator_base;
+    //!\}
+
+    // Import the base::operator()
+    using file_validator_base::operator();
+
+    /*!\brief Tests whether path is an existing regular file and is readable.
+     * \param file The input value to check.
+     * \throws parser_invalid_argument if the validation process failed or std::filesystem::filesystem_error if some
+     *         unexpected errors during the filesystem operations occur.
+     */
+    void operator()(std::filesystem::path const & file) const override
+    {
+        if (!std::filesystem::exists(file))
+            throw parser_invalid_argument(detail::to_string("The file ", file, " does not exist!"));
+
+        // Check if file has any read permissions.
+        if (!path_is_readable(file))
+            throw parser_invalid_argument(detail::to_string("Cannot read ", file, "!"));
+
+        // Check extension.
+        file_validator_base::validate_filename_impl(file);
```

Comment:
Do you need the `file_validator_base::`? It's simple inheritance or not?

Reply:
Not sure, but I find it good style, since I see, where I need to look for the function definition/declaration.